db/db_impl/db_impl_write.cc

```diff
@@ -854,8 +854,9 @@ void DBImpl::WriteStatusCheck(const Status& status) {
 void DBImpl::IOStatusCheck(const IOStatus& io_status) {
   // Is setting bg_error_ enough here?  This will at least stop
   // compaction and fail any further writes.
-  if (immutable_db_options_.paranoid_checks && !io_status.ok() &&
-      !io_status.IsBusy() && !io_status.IsIncomplete()) {
+  if ((immutable_db_options_.paranoid_checks && !io_status.ok() &&
```

Comment:
We are not adding this check to `WriteStatusCheck` because we expect this to occur only for `io_status`? If so, can we add an assertion?

Reply:
Yes, any IO errors are supposed to be handled based on the ```IOStatus```.  If ```WriteStatusCheck``` is called for an IO error, ```SetBGError``` should have already been called for it. I'll add an assertion for it.